.github/workflows/build.yml

```diff
@@ -42,6 +42,78 @@ jobs:
             -Psonatype.username=${{ secrets.SONATYPE_USERNAME }}
             -Psonatype.password=${{ secrets.SONATYPE_PASSWORD }}
 
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Unit tests
+        run: ./gradlew testDebugUnitTest
+        working-directory: .
+      - name: Upload failed unit test artifacts
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-failures
+          path: '**/build/reports'
+
+  instrumentation-tests:
```

Comment:
We might want to make this depend on the `build` job running successfully, as it can take quite a lot longer than the other jobs (and I believe you can only use on MacOS agent at a time - i.e. multiple PRs would have to wait for the agent to be free). 

However that would make the PR take slightly longer to be ready

Reply:
`build` depends on `test*UnitTest` and `lint` tasks, you can launch locally to verify. No need to have a separate config for it.
Please just add `actions/upload-artifact@v3` to the main config.